spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
```

Comment:
Yep, option 3. is the best. There's already a signal called `sig_pythonpath_changed` in the main window. Please see if that's what you need.